### PR TITLE
refactor(web): Allow dropdown for more general use

### PR DESCRIPTION
**Pull request by danieldietzler**
Now, options can have a generic type. A render function can specify what text (and icon) should be displayed for an option.
Also some design improvements by making the dropdown a little more compact:

Preview is available [here](https://pr-4515.preview.immich.app)

### CODE REVIEW

File: web/src/lib/components/elements/dropdown.svelte [hunk 1]
```diff
@@ -50,22 +53,22 @@
       transition:fly={{ y: -30, x: 30, duration: 200 }}
       class="text-md absolute right-0 top-5 z-50 flex min-w-[250px] flex-col rounded-2xl bg-gray-100 py-4 text-black shadow-lg dark:bg-gray-700 dark:text-white"
     >
-      {#each options as option, index (option)}
```
Comment (alextran1502) [inline web/src/lib/components/elements/dropdown.svelte:53]: if you remove `(option)`, any changes to the `options` list won't trigger the re-render 

reference: https://learn.svelte.dev/tutorial/keyed-each-blocks

Comment (danieldietzler) [reply web/src/lib/components/elements/dropdown.svelte:53]: Ah ok thanks! I thought it was related to index oops

**Review by jrasm91 — APPROVED** [review]
Lgtm

**Comment by alextran1502** [comment]
Hello, can you help revert the change to the alignment of the text without the icon? It is the design intention

**Comment by danieldietzler** [comment]
> Hello, can you help revert the change to the alignment of the text without the icon? It is the design intention

@alextran1502 I've reverted the design changes so that it looks as before. Feel free to have a look at the preview again to verify

**Comment by waclaw66** [comment]
Causes issue https://github.com/immich-app/immich/issues/4907

**Pull request merged**